frame/vault/src/tests.rs

```diff
@@ -525,93 +525,125 @@ proptest! {
-				prop_assert_ok!(Vaults::withdraw(Origin::signed(account), vault_id, lp));
-
-				// Balance after having deposited + withdrawn my funds
-				let new_balance = Tokens::balance(asset_id, &account);
-
-				// We had shares before the profit, we get a cut of the profit
-				if index <= strategy_deposit_moment {
-					// Compute my share
-					let strategy_profit_share =
-						multiply_by_rational(strategy_profits, lp, lp_total).expect("qed;");
-
-					prop_assert_epsilon!(new_balance, balance + strategy_profit_share);
-				}
-				else {
+			for (idx, (account, initial_native_tokens)) in lps.iter().copied().enumerate() {
+				//  Contains half of LP balances minus LP profits
+				let half_initial_native_tokens = initial_native_tokens / 2;
+
+				let lp_tokens = Tokens::balance(vault.lp_token_id, &account);
+				// Because of `<Vaults as StrategicVault>::withdraw`, the vault does not own 100% of
+				// the funds. Therefore, a full withdraw is not possible.
+				let withdrawn_lp_tokens = lp_tokens / 2;
+
+				// Withdraws all LP tokens
+				prop_assert_ok!(Vaults::withdraw(Origin::signed(account), vault_id, withdrawn_lp_tokens));
+
+				// New balance that includes losses and profits
+				let new_native_tokens = Tokens::balance(asset_id, &account);
+
+				let curr_lp_deposited_before_moment = lps_start + idx <= strategy_moment;
+
+				if curr_lp_deposited_before_moment {
+					let strategy_native_tokens_deposit = multiply_by_rational(
+						strategy_deposit / 2,
+						lp_tokens,
+						lp_tokens_total,
+					)
+					.expect("qed;");
+					let strategy_native_tokens_withdraw = multiply_by_rational(
+						strategy_withdraw_diff / 2,
+						lp_tokens,
+						lp_tokens_total,
+					)
+					.expect("qed;");
+
+					let diff = strategy_native_tokens_deposit - strategy_native_tokens_withdraw;
+
+					prop_assert_epsilon!(new_native_tokens, half_initial_native_tokens + diff);
+				} else {
 					// Our balance should be equivalent
-					prop_assert_epsilon!(new_balance, balance);
+					prop_assert_epsilon!(new_native_tokens, half_initial_native_tokens);
 				}
 			}
 
 			// Global check
-			let shareholders = &accounts[0..strategy_deposit_moment];
-			let initial_sum_of_shareholders_balance = shareholders.iter()
-				.map(|(_, initial_balance)| initial_balance)
+			let initial_sum_of_native_tokens = before_moment_lps()
+				.map(|(_, initial_native_tokens)| initial_native_tokens)
 				.sum::<Balance>();
-			let current_sum_of_shareholders_balance = shareholders.iter()
-				.map(|(account, _)| Tokens::balance(asset_id, account))
+			let current_sum_of_native_tokens = before_moment_lps()
+				.map(|(account, _)| Tokens::balance(asset_id, &account))
 				.sum::<Balance>();
 
 			prop_assert_epsilon!(
-				current_sum_of_shareholders_balance,
-				initial_sum_of_shareholders_balance + strategy_profits
+				current_sum_of_native_tokens,
+				initial_sum_of_native_tokens / 2 + strategy_deposit / 2 - strategy_withdraw_diff / 2
```

Comment:
Because LPs are taking half of their LP tokens (https://github.com/ComposableFi/composable/pull/299/files#diff-a8b4bcca53a49de6c5fa8f929af30fbecd3773c39c5fdbec31b6b495eb433931R601-R603), this and other divisions by 2 are necessary